aim/aim_manager.py

```diff
@@ -203,7 +203,7 @@ def update(self, context, resource, fix_ownership=False,
                     if old_resource.user_equal(
```

Comment:
It seems to me that old_resource_copy will have all the fields available including this new field 'mode' as it was built from DB in line# 191. If that's the case then we don't really need the new API "def attributes_equivalent(superset, subset)" defined in resource.py.

Reply:
I'm not sure I follow. The new code checks if it's a subset or not. Without it, an update request that leaves out the mode field would result in a new version of the object with an epoch bump, even though the objects are equivalent.

Reply:
I think you meant if update_attr_val doesn't have the 'mode' attribute in it. But in line#202, you can see that the code applies those update to the old_resource variable which was built in line#191 from db_obj which is from DB. So old_resource should already have that 'mode' field even though the update_attr_val doesn't have it. Make sense?

Reply:
The problem is that the setattr is setting the entire static_path attribute using what was provided by the caller - not updating the individual dicts in the static_paths list.

Reply:
Hmm......Wondering how would that work before if an update request leaving out the 'encap' field comes in?

Reply:
The model class won't create the associated entry in the DB, which means it won't get added to the list:
https://github.com/noironetworks/aci-integration-module/blob/master/aim/db/models.py#L342

That's the existing behavior (i.e. nothing changes with this patch).

Reply:
Then I saw u added "mode=p.get('mode', 'regular')" in line#346 of models.py, that should be good enough right? It should be a similar problem if an update request leaving out the 'host' field comes in, how did we handle that before then?

Reply:
If host isn't set, it will set it to an empty string in the model class, but neither host or mode are requirements for creating a new entry in the DB (i.e. they don't appear in that if statement).

It's worth pointing out that this method suffers from the same issue that the security groups/rules suffered - namely that the list of DB entries are deleted and recreated each time (fixed in https://github.com/noironetworks/aci-integration-module/commit/3b1fd190293569f4e82902cf69322374e143a3cc#diff-27d21d9e0ec23b43a0e3f5d38ef04922). We probably should revisit all our objects that have lists and fix any that are doing the same thing.